indico/modules/events/contributions/controllers.py

```diff
@@ -95,8 +78,40 @@ class RHContributions(RHManageContributionsBase):
     """Display contributions management page"""
 
     def _process(self):
+        contrib_report_config = reporter.get_report_config(self.event_new.id)
+        if 'config' in request.args:
+            return redirect(url_for('.manage_contributions', self.event_new))
+
+        contrib_report_args = reporter.get_contrib_report_args(self.event_new, contrib_report_config['filters'])
+
         return WPManageContributions.render_template('management/contributions.html', self._conf, event=self.event_new,
-                                                     **_get_contribution_list_args(self.event_new))
+                                                     **contrib_report_args)
+
+
+class RHContributionsListCustomize(RHManageContributionsBase):
+    """Filter options for the contribution list of an event"""
+
+    def _process_GET(self):
+        contrib_report_config = reporter.get_report_config(self.event_new.id)
+        filterable_items = reporter.FILTERABLE_ITEMS
+        filterable_items['session']['filter_choices'] = {unicode(s.id): s.title
```

Comment:
You are modifying global state here! It might cause some ugly issues if someone was using a multithreaded webserver. Either `deepcopy` the filterable items or, if you need the proper list in the class to, maybe creating an instance is a good idea so you can store it on the instance (also also move this code here in the class, e.g. in the constructor which would then take the event as an argument)